### PR TITLE
[clangd] Propagate context into stdlib indexing thread

### PR DESCRIPTION
**Pull request by kadircet**
Some FS implementations rely on snapshots available in the context.

### CODE REVIEW

File: clang-tools-extra/clangd/ClangdServer.cpp [hunk 1]
```diff
@@ -112,7 +112,10 @@ struct UpdateIndexCallbacks : public ParsingCallbacks {
                  // Index outlives TUScheduler (declared first)
                  FIndex(FIndex),
                  // shared_ptr extends lifetime
-                 Stdlib(Stdlib)]() mutable {
+                 Stdlib(Stdlib),
+                 // We have some FS implementations that rely on infomration in
```
Comment (ilya-biryukov) [inline clang-tools-extra/clangd/ClangdServer.cpp:116]: Drive-by comment: there's a typo here.

s/infomration/information

File: clang-tools-extra/clangd/ClangdServer.cpp [hunk 2]
```diff
@@ -112,7 +112,10 @@ struct UpdateIndexCallbacks : public ParsingCallbacks {
...
+                 // the context.
+                 Ctx(Context::current().clone())]() mutable {
```
Comment (sam-mccall) [inline clang-tools-extra/clangd/ClangdServer.cpp:118]: this doesn't work - need to actually WithContext it inside the body

**Review by sam-mccall — COMMENTED** [review]
I'm iffy on whether this is semantically right or not, but we need to solve this problem somehow, so let's try the most obvious way.

**Review by sam-mccall — APPROVED** [review]
